target/riscv/cpu_helper.c

```diff
@@ -608,13 +608,22 @@ static int get_physical_address(CPURISCVState *env, hwaddr *physical,
         } else if (ppn & ((1ULL << ptshift) - 1)) {
             /* Misaligned PPN */
             return TRANSLATE_FAIL;
+#if defined(TARGET_CHERI) && !defined(TARGET_RISCV32)
         } else if ((access_type == MMU_DATA_LOAD ||
                     access_type == MMU_DATA_CAP_LOAD) &&
+#else
+        } else if (access_type == MMU_DATA_LOAD &&
+#endif
```

Comment:
```suggestion
        } else if ((access_type == MMU_DATA_LOAD
#if defined(TARGET_CHERI) && !defined(TARGET_RISCV32)
                   || access_type == MMU_DATA_CAP_LOAD
#endif
                   ) &&
```
etc everywhere is uglier but probably the better thing to do.